src/dune_rules/dune_project.ml

```diff
@@ -692,9 +692,10 @@ let encode : t -> Dune_lang.t list =
       [ string "lang"; string "dune"; Dune_lang.Syntax.Version.encode dune_version ]
   in
   let dialects =
-    if Dialect.DB.is_default dialects
-    then []
-    else Dialect.DB.fold ~f:(fun d ls -> Dialect.encode d :: ls) ~init:[] dialects
```

Comment:
Instead of the `dialects` value into an option, why not simply remove `is_default` and the `then` case here? As far as I can see the `builtin` values appear in the result of this function as soon as there is at least one extra dialect.